projects/aqlprofile/src/core/pm4_factory.cpp

```diff
@@ -52,14 +56,34 @@ locked_agent_cache& get_cache() {
 }  // namespace
 
 aqlprofile_agent_handle_t RegisterAgent(const aqlprofile_agent_info_v1_t* agent_info) {
-  aqlprofile_agent_handle_t agent_id;
-  AgentInfo int_agent_info;
-  int_agent_info.cu_num = agent_info->cu_num;
-  int_agent_info.se_num = agent_info->se_num;
-  int_agent_info.xcc_num = agent_info->xcc_num;
-  int_agent_info.shader_arrays_per_se = agent_info->shader_arrays_per_se;
-  int_agent_info.domain = agent_info->domain;
-  int_agent_info.bdf_id = agent_info->location_id;
+  // If assertion fails,
+  // Update handling of new fields
+  // Update size to new size
+  static_assert(sizeof(aqlprofile_agent_info_v1_t) == 40);
+
+  aqlprofile_agent_handle_t agent_id{};
+  AgentInfo int_agent_info{};
+
+  if (agent_info->size == 40) {
```

Comment:
Id invert this if statement to reduce indentation. 


```suggestion
  if (agent_info->size != 40) {
      //error
  }
  
  // copy data.
```